File: desktop-ui/src/main/java/com/bgconsole/desktop_ui/workspace/WorkspaceService.java
package com.bgconsole.desktop_ui.workspace;

import com.bgconsole.desktop_ui.location.Location;

public interface WorkspaceService {

    Workspace createWorkspace(String profileId, String name, String path);

    Workspace createLocalWorkspace(String workspaceId);

    Workspace loadWorkspace(String path);

    Workspace createCommandList(Location location, String name);

    Workspace createCommandListLocally(Location location, String name);

}

File: core/src/main/kotlin/com/bgconsole/core/workspace/WorkspaceService.kt
package com.bgconsole.core.workspace

import com.bgconsole.core.common.CRUDService
import com.bgconsole.domain.Location
import com.bgconsole.domain.Workspace

interface WorkspaceService : CRUDService<Workspace> {

    fun findByLocation(location: Location): Workspace
}